src/Functions/test/Functions.Scenario.Tests.ps1

```diff
@@ -249,119 +302,425 @@ Describe 'Functions End to End Tests' {
+                "DotNet" = "3"
+                "Python" = "3.7"
+                "Java" = "8"
+            }
+        }
+        "Windows" = @{
+            "2"= @{
+                "Node" = "10"
+                "DotNet"= "2"
+                "PowerShell" = "6"
+                "Java" = "8"
+            }
+            "3" =  @{
+                "Node" = "10"
+                "DotNet" = "3"
+                "PowerShell" = "6"
+                "Java" = "8"
+            }
+        }
+    }
+
+    $LinuxRuntimes = @("DotNet", "Node", "Java", "Python")
+    $WindowsRuntimes = @("DotNet", "Node", "Java", "PowerShell")
+
+    $LinuxTestData = @{
+        "PlanName" = $planNameWorkerTypeLinux
+        "Runtimes" = $LinuxRuntimes
+        "StorageAccountName" = $storageAccountLinux
+        "resourceGroupName" = $resourceGroupNameLinuxPremium
+        "Location" = $linuxConsumptionLocation
+    }
+    $WindowsTestData = @{
+        "PlanName" = $planNameWorkerTypeWindows
+        "Runtimes" = $WindowsRuntimes
+        "StorageAccountName" = $storageAccountWindows
+        "resourceGroupName" = $resourceGroupNameWindowsPremium
+        "Location" = $windowsConsumptionLocation
+    }
+
+    function GetTestData($OSType)
+    {
+        if ($OSType -eq "Linux")
+        {
+            return $LinuxTestData
+        }
+        else
+        {
+            return $WindowsTestData
+        }
+    }
+
+    $filePath = Join-Path $PSScriptRoot "warningOutput.log"
+    $null = New-Item -Path $filePath -ItemType File -Force
+
+    foreach ($OSType in @("Linux", "Windows"))
+    {
+        foreach ($functionsVersion in @("2", "3"))
+        {
+            $testData = GetTestData -OSType $OSType
+
+            $location = $testData["Location"]
+            $storageAccountName = $testData["StorageAccountName"]
+            $runtimes = $testData["Runtimes"]
+            $resourceGroupName = $testData["resourceGroupName"]
+
+            foreach ($runtime in $runtimes)
             {
-                break
+                if (($functionsVersion -eq "2") -and ($OSType -eq "Linux") -and ($runtime -eq "Java"))
+                {
+                    # Java 8 is not supported in Linux for Functions V2.
+                    continue
+                }
+
+                It "Validate New-AzFunctionApp default runtime version for $runtime in Functions version $functionsVersion for $OSType" {
+
+                    # Note: These set of tests are for consumptions function apps. We do this for two things:
+                    # 1) Test case is faster, we do not need to validate the service plan name
+                    # 2) Validate the -Location code path
+                    # We use -WhatIf which performs all the inputs validation for the function app creation, and we return right before sending the request to the backend
+
+                    try
+                    {
+                        $functionName = "Func99-$OSType-$runtime-" + (Get-Random).ToString()
```

Comment:
Seems like this will cause problems for playback, as the function name will differ for each iteration

Reply:
Hi @markcowl, I removed all calls to `get-random` to ensure test can run playback mode. Could you please take another look? Thanks!